src/components/ebay-confirm-dialog/examples/default.marko

```diff
@@ -38,18 +39,21 @@ $ const {
 
 <ebay-button on-click('openDialog')>Open Confirm Dialog</ebay-button>
 class {
-    onInput(input) {
+    declare state: {
+        open?: boolean;
+    }
+    onInput(input: Input) {
         this.state = { open: input.open };
     }
     openDialog() {
         this.state.open = true;
     }
-    closeDialog(ev) {
+    closeDialog() {
         this.state.open = false;
-        this.emit('reject', ev);
+        this.emit('reject', ...arguments);
```

Comment:
I'm not sure if we should recommend that people do this, since `arguments` isn't well-defined. Also, you don't have `on-reject` or `onReject` typed in `Input` so people wouldn't be able to use this component in a project.

Reply:
I thought this emit is just for storybook actions to work (not for demo purpose), so I simplified it to avoid type error. I can refine it not big deal, only like 4-5 places.

Reply:
Yeah I think it's worth refining. A lot of people copy-paste from these examples so they should use best practices.